Source/WebCore/Modules/websockets/WorkerThreadableWebSocketChannel.cpp

```diff
@@ -414,9 +414,12 @@ void WorkerThreadableWebSocketChannel::Bridge::connect(const URL& url, const Str
 
         auto& document = downcast<Document>(context);
         
-        if (document.frame() && !MixedContentChecker::frameAndAncestorsCanRunInsecureContent(*document.frame(), document.securityOrigin(), url, MixedContentChecker::ShouldLogWarning::No)) {
-            peer->fail(makeString("The page at ", document.url().stringCenterEllipsizedToLength(), " was blocked from connecting insecurely to ", url.stringCenterEllipsizedToLength(), " either because the protocol is insecure or the page is embedded from an insecure page."));
-            return;
+        if (auto* frame = document.frame()) {
+            if (!MixedContentChecker::frameAndAncestorsCanRunInsecureContent(*frame, document.securityOrigin(), url, MixedContentChecker::ShouldLogWarning::No)
```

Comment:
If all our calls to frameAndAncestorsCanRunInsecureContent are complemented with a shouldBlockInsecureContent call, we might want to do the shouldBlockInsecureContent inside frameAndAncestorsCanRunInsecureContent.

Reply:
There is one instance in CachedResourceLoader::checkInsecureContent where we don't want to consider shouldBlockInsecureContent. I considered creating a wrapper method in MixedContentChecker that returns `!frameAndAncestorsCanRunInsecureContent || shouldBlockInsecureContent`. I'll investigate different patterns.